CDCgov/prime-reportstream
# Update Condition Filter to use OR instead of AND

This PR updates the Condition filters to pass items if any of the filters matches the report's data.

Note - This PR has attached a CURL call that I used on Postman to test. Feel free to use it or make your own test.

Test Steps:
1. In organizations.yml, update the `Ignore FULL_ELR` receiver to have the following condition filter, replacing 600-7 with whatever LOINC code shows up in your Postman test:
`conditionFilter: ["%obsPerformedCodes.intersect('600-7').exists()"]`
2. Run ReportStream, then run`./gradlew reloadTables` and `./gradlew reloadSettings` to update the settings.
3. Submit a report with data that would go to the `Ignore FULL_ELR` Receiver.
4. Using the History API, verify that the report's items were successfully delivered to `Ignore FULL_ELR`.
5. Repeat steps 1-4, but change the `conditionFilter`'s LOINC code to something that is not in the submitted report (e.g.`999-9`). The item should now be filtered out.
6. Add multiple values to `conditionFilter`, with at least one that matches the submission data, and at least one that does not matches. for example,
```
conditionFilter:
        - "%obsPerformedCodes.intersect('999-9').exists()"
        - "%obsPerformedCodes.intersect('600-7').exists()"
        - "%obsPerformedCodes.intersect('123-4').exists()"
```

## Changes
- Update logic on the FHIRRouter to check condition filters in an OR manner instead of AND.

## Checklist

## Linked Issues
- #8837 

[local call curl.txt](https://github.com/CDCgov/prime-reportstream/files/11466673/local.call.curl.txt)
